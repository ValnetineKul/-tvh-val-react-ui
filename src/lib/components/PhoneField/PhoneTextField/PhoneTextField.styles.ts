import { makeStyles } from '../../../../themes/core';

const useStyles = makeStyles({ name: 'PhoneField' })((theme) => ({
  root: {
    '& label': {
      paddingLeft: `calc(${theme.layoutSpacing['spacing/550']} + ${theme.layoutSpacing['spacing/300']}) !important`,
    },
  },
  rootReadOnly: {
    '& label': {
      paddingLeft: `${theme.layoutSpacing['spacing/500']} !important`,
    },
  },
  input: {
    paddingLeft: `${theme.layoutSpacing['spacing/600']} !important`,
  },
  inputReadOnly: {
    paddingLeft: `${theme.layoutSpacing['spacing/525']} !important`,
  },
  inputElement: {
    paddingLeft: `${theme.layoutSpacing['spacing/000']} !important`,
  },
}));

export default useStyles;
