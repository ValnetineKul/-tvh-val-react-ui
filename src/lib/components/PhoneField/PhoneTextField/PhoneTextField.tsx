import React, { forwardRef } from 'react';

import TextField from '../../TextField';
import useStyles from './PhoneTextField.styles';

// This is required since TS needs named type to export as .d.ts file.
// Point being, we are npm package now and need to export TS typings.
export interface PhoneTextFieldProps extends React.ComponentProps<typeof TextField> {}

const PhoneTextField = forwardRef<HTMLInputElement, PhoneTextFieldProps>((props, ref) => {
  const { classes, cx } = useStyles();

  return (
    <TextField
      {...props}
      inputRef={ref}
      className={cx(classes.root, { [classes.rootReadOnly]: !!props.readOnly })}
      inputClassName={props.readOnly ? classes.inputReadOnly : classes.input}
      inputElementClassName={classes.inputElement}
    />
  );
});
PhoneTextField.displayName = 'PhoneTextField';

export default PhoneTextField;
