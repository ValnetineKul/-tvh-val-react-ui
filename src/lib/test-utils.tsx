import userEvent from '@testing-library/user-event';
import { fireEvent, render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { theme } from '../themes/tvh';
import { createTheme, ThemeProvider, SurfaceProvider } from '../themes/core';

export const muiTheme = createTheme(theme);

const AllTheProviders: React.FC = ({ children }) => {
  return (
    <MemoryRouter>
      <ThemeProvider theme={muiTheme}>
        <SurfaceProvider color="100">{children}</SurfaceProvider>
      </ThemeProvider>
    </MemoryRouter>
  );
};

const customRender = (component: React.ReactElement, options?: Parameters<typeof render>[1]) =>
  render(component, { wrapper: AllTheProviders, ...options });

const customRenderHook: typeof renderHook = (hook, options) =>
  renderHook(hook, { wrapper: AllTheProviders, ...options });

export const initResponsiveTest = (size: number | 'Mobile' | 'Tablet' | 'Desktop' | 'LgDesktop' | 'XlDesktop') => {
  let width: number;
  if (size === 'Mobile') width = muiTheme.breakpoints.values.sm - 1;
  else if (size === 'Tablet') width = muiTheme.breakpoints.values.sm;
  else if (size === 'Desktop') width = muiTheme.breakpoints.values.md;
  else if (size === 'LgDesktop') width = muiTheme.breakpoints.values.lg;
  else if (size === 'XlDesktop') width = muiTheme.breakpoints.values.xl;
  else width = size;

  Object.defineProperty(window, 'innerWidth', {
    writable: true,
    configurable: true,
    value: width,
  });

  window.matchMedia = jest.fn().mockImplementation((query) => {
    const queryWidthInterval: string[] = query.match(/\d+(\.\d+)?/g);

    let matches;
    if (queryWidthInterval.length === 2) {
      matches = width >= +queryWidthInterval[0] && width < +queryWidthInterval[1];
    } else {
      const queryWidth = +queryWidthInterval[0];
      const isDown = /max/.test(query);

      matches = isDown ? width <= queryWidth : width >= queryWidth;
    }

    return {
      matches,
      media: query,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
    };
  });
};

const customUserEvent = Object.assign(userEvent, {
  mouseDown: fireEvent.mouseDown,
  mouseUp: fireEvent.mouseUp,
  mouseMove: fireEvent.mouseMove,
  scroll: fireEvent.scroll,
  error: fireEvent.error,
});

export { screen, within, waitFor, act, waitForElementToBeRemoved } from '@testing-library/react';
// override render method
export { customRender as render, customRenderHook as renderHook, customUserEvent as userEvent };
