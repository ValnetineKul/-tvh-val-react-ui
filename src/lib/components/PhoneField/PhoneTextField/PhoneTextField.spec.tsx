import React from 'react';
import { render } from '../../../test-utils';
import PhoneTextField from './PhoneTextField';

describe('PhoneTextField', () => {
  it('Should render correctly', () => {
    const { container } = render(<PhoneTextField readOnly />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
