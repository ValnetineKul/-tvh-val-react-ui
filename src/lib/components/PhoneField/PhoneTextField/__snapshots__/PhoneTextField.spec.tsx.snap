// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneTextField Should render correctly 1`] = `
<div
  class="MuiFormControl-root tss-10-TextField-readOnly-ref tss-6bzbxe-TextField-root-PhoneField-root-PhoneField-rootReadOnly-TextField-readOnly mui-1nrlq1o-MuiFormControl-root"
>
  <div
    class="tss-bdw984-Surface-root"
  >
    <div
      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl tss-10-TextField-inputRoot-ref tss-mptm0n-TextField-inputRoot-PhoneField-inputReadOnly mui-183rr2a-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        class="tss-ajocaw-TextField-input-PhoneField-inputElement MuiOutlinedInput-input MuiInputBase-input mui-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        readonly=""
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="tss-tyiogj-TextField-notchedOutline tss-10-TextField-notchedOutline-ref MuiOutlinedInput-notchedOutline mui-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="mui-nnbavb"
        >
          <span
            class="notranslate"
          >
            â€‹
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    class="tss-19z6szl-TextField-helpers"
  />
</div>
`;
